vignettes/package_structure.Rmd

```diff
@@ -0,0 +1,162 @@
+---
+title: "Package Structure"
+package: mmrm
+author:
+  - name: Liming Li
+output:
+  rmarkdown::html_document:
+          theme: "spacelab"
+          highlight: "kate"
+          toc: true
+          toc_float: true
+vignette: |
+  %\VignetteIndexEntry{Contributing Guide for Developers}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options:
+  chunk_output_type: console
+---
+
+# Introduction
+
+As `mmrm` package is built on `TMB` with complex structure, it might be hard for new developers to onboard.
+To make it easier for new developers to join, this documentation is created.
+
+# Package Structures
+
+`mmrm` is an R package, and the following sections describe all files/folders that are contained in this package.
+
+## data
+
+The `data/` folder is where `mmrm` internal data is stored.
+Currently, we only have `fev_data.rda` in this folder.
+In turn, `fev_data.rda` only contains the `fev_data` dataset.
+
+`fev_data` is a data frame of the following structure:
+
+```{r}
+library(mmrm)
+head(fev_data)
+```
+
+`USUBJID` is the unique subject id(total 200 subjects), `AVISIT` is the analysis visit(from VIS1 to VIS4), `TRT` is the treatment assignment,
+`RACE` is the race of subject, `SEX` is the gender of subject, `FEV1` is the forced expiratory volume, `FEV1_BL` is the baseline forces expiratory volume.
+`WEIGHT` is the weight for each observation, `VISITN` is the numeric value of `AVISIT` (taking the prefix "VIS" off from AVISIT), `VISITN2` is a numeric value.
+
+In `mmrm` analysis, we usually use `FEV1` as the response, `AVISIT` as the time variable, and `USUBJID` as subject variable.
+See [introduction](introduction.html#common-usage).
```

Comment:
this seems duplicate with the documentation for `fev_data`, suggest to skip and if something missing from the data documentation add it there